### PR TITLE
Update configuration-settings.rst

### PR DESCRIPTION
**Pull request by wiersgallak**
Clarified the requirement of email verification configuration must be set for restrictions to be applied.

### CODE REVIEW

File: source/configure/configuration-settings.rst [hunk 1]
```diff
@@ -2806,7 +2806,7 @@ Restrict account creation to specified email domains
 
 Teams and user accounts can only be created by a verified email from this list of comma-separated domains (e.g. "corp.mattermost.com, mattermost.org").
 
-This setting only affects email login.
+This setting only affects email login. For domains to be restricted, you must also set Require Email Verification to ``True``.
```
Comment (cwarnermm) [inline source/configure/configuration-settings.rst:2809]:
```suggestion
This setting only affects email login. For domains to be restricted, you must also set `Require Email Verification <https://docs.mattermost.com/configure/configuration-settings.html#require-email-verification>`__ to ``true``. 
```

File: source/messaging/team-settings.rst [hunk 2]
```diff
@@ -49,9 +49,9 @@ You can upload a team icon in BMP, JPG or PNG format. Square images with a solid
 Allow only users with a specific email domain to join this team
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Specifying domains in this setting prevents users from joining the team when their email domain is not on the list. Only users that have an email domain matching the defined domains may join the team.
+Specifying domains in this setting prevents users from joining the team when their email domain is not on the list. Only users that have an email domain matching the defined domains may join the team. 
 
-Users without a matching domain on the team prior to the domain being specified will not be removed after the domain is added.
+Users without a matching domain on the team prior to the domain being specified will not be removed after the domain is added. For domains to be restricted, **System Console > Authentication > Email > Require Email Verification** must be set to ``True``.
```
Comment (cwarnermm) [inline source/messaging/team-settings.rst:54]:
```suggestion
Users without a matching domain on the team prior to the domain being specified will not be removed after the domain is added. For domains to be restricted, **System Console > Authentication > Email > Require Email Verification** must be set to ``true``. 
```

File: source/configure/configuration-settings.rst (hunk 1 above)
Comment (cwarnermm) [reply source/configure/configuration-settings.rst:2809]: Added a direct link to the config setting

**Review by cwarnermm — COMMENTED** [review]
Thanks! I've provided feedback inline.

File: source/messaging/team-settings.rst [hunk 3]
```diff
@@ -49,9 +49,9 @@ You can upload a team icon in BMP, JPG or PNG format. Square images with a solid
 Allow only users with a specific email domain to join this team
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Specifying domains in this setting prevents users from joining the team when their email domain is not on the list. Only users that have an email domain matching the defined domains may join the team.
+Specifying domains in this setting prevents users from joining the team when their email domain is not on the list. Only users that have an email domain matching the defined domains may join the team. 
 
-Users without a matching domain on the team prior to the domain being specified will not be removed after the domain is added.
+Users without a matching domain on the team prior to the domain being specified will not be removed after the domain is added. For domains to be restricted, **System Console > Authentication > Email > Require Email Verification** must be set to ``true``.
```
Comment (DSchalla) [inline source/messaging/team-settings.rst:54]:
```suggestion
Users without a matching domain on the team prior to the domain being specified will not be removed after the domain is added. For servers with email authentication **System Console > Authentication > Email > Require Email Verification** must be set to ``true`` for domain restrictions to be effective.
```

File: source/configure/configuration-settings.rst [hunk 4]
```diff
@@ -2806,7 +2806,7 @@ Restrict account creation to specified email domains
 
 Teams and user accounts can only be created by a verified email from this list of comma-separated domains (e.g. "corp.mattermost.com, mattermost.org").
 
-This setting only affects email login.
+This setting only affects email login. For domains to be restricted, you must also set `Require Email Verification <https://docs.mattermost.com/configure/configuration-settings.html#require-email-verification>`__ to ``true``.
```
Comment (DSchalla) [inline source/configure/configuration-settings.rst:2809]:
```suggestion
This setting only affects email login. For domains restrictions to be effective, you must also set `Require Email Verification <https://docs.mattermost.com/configure/configuration-settings.html#require-email-verification>`__ to ``true``. 
```

**Review by DSchalla — CHANGES_REQUESTED** [review]
Two minor change proposals, LGTM otherwise

**Review by cwarnermm — APPROVED** [review]
Thanks!

**Review by DSchalla — APPROVED** [review]

File: source/configure/configuration-settings.rst [hunk 5]
```diff
@@ -2806,7 +2806,7 @@ Restrict account creation to specified email domains
 
 Teams and user accounts can only be created by a verified email from this list of comma-separated domains (e.g. "corp.mattermost.com, mattermost.org").
 
-This setting only affects email login.
+This setting only affects email login. For domains restrictions to be effective, you must also set `Require Email Verification <https://docs.mattermost.com/configure/configuration-settings.html#require-email-verification>`__ to ``true``.
```
Comment (justinegeffen) [inline source/configure/configuration-settings.rst:2809]:
```suggestion
This setting only affects email login. For domain restrictions to be effective, you must also set `Require Email Verification <https://docs.mattermost.com/configure/configuration-settings.html#require-email-verification>`__ to ``true``.
```

File: source/messaging/team-settings.rst [hunk 6]
```diff
@@ -49,9 +49,9 @@ You can upload a team icon in BMP, JPG or PNG format. Square images with a solid
 Allow only users with a specific email domain to join this team
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Specifying domains in this setting prevents users from joining the team when their email domain is not on the list. Only users that have an email domain matching the defined domains may join the team.
+Specifying domains in this setting prevents users from joining the team when their email domain is not on the list. Only users that have an email domain matching the defined domains may join the team.
```
Comment (justinegeffen) [inline source/messaging/team-settings.rst:52]:
```suggestion
Specifying domains in this setting prevents users from joining the team when their email domain is not on the list. Only users that have an email domain matching the defined domains may join the team.
```

File: source/messaging/team-settings.rst [hunk 7]
```diff
@@ -49,9 +49,9 @@ You can upload a team icon in BMP, JPG or PNG format. Square images with a solid
 Allow only users with a specific email domain to join this team
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Specifying domains in this setting prevents users from joining the team when their email domain is not on the list. Only users that have an email domain matching the defined domains may join the team.
+Specifying domains in this setting prevents users from joining the team when their email domain is not on the list. Only users that have an email domain matching the defined domains may join the team. 
 
-Users without a matching domain on the team prior to the domain being specified will not be removed after the domain is added.
+Users without a matching domain on the team prior to the domain being specified will not be removed after the domain is added. For servers with email authentication **System Console > Authentication > Email > Require Email Verification** must be set to ``true`` for domain restrictions to be effective.
```
Comment (justinegeffen) [inline source/messaging/team-settings.rst:54]:
```suggestion
Users without a matching domain on the team prior to the domain being specified will not be removed after the domain is added. For servers with email authentication **System Console > Authentication > Email > Require Email Verification** must be set to **true** for domain restrictions to be effective.
```

**Review by justinegeffen — COMMENTED** [review]
Some minor corrections prior to merge. :)

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**